geocity/apps/api/views.py

```diff
@@ -546,6 +546,23 @@ def image_thumbor_display(request, submission_id, image_name):
     return thumbor_response
 
 
+def get_agenda_submissions(entities, submissions):
+    # To validate a request and show it in agenda, an user need to be pilot of his own entity and validator for other entities.
+    # Retrieve pilots of entity
+    pilot_of_entity = User.objects.filter(
+        groups__permit_department__administrative_entity__in=entities,
+        groups__permit_department__is_backoffice=True,
+    ).values("id")
+
+    # Check agenda submissions is validated by any user on the pilot group of it's own entity
+    submissions = submissions.filter(
+        Q(administrative_entity__in=entities)
+        | Q(validations__validated_by__in=pilot_of_entity)
+    )
+
+    return submissions
+
+
```

Comment:
Is this not generating too many SQL queries ?

Reply:
![image](https://github.com/yverdon/geocity/assets/36194606/b92329a0-2019-4fa9-b4fa-645c5667df12)

Reply:
avec 60 éléments : 
![image](https://github.com/yverdon/geocity/assets/36194606/5f41428f-c3e8-4733-a613-72abbf00f88e)

Reply:
Avec 600 événements dans 6 entités différentes :
![image](https://github.com/yverdon/geocity/assets/36194606/18136319-bb41-4dee-a316-ed8a156062f6)

Reply:
Probablement optimisable :

![image](https://github.com/yverdon/geocity/assets/36194606/a3c399fd-abbc-4d8f-91a9-d4333ee7c8dc)